compiler/rustc_resolve/src/late.rs

```diff
@@ -4062,14 +4064,21 @@ impl<'a, 'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
         source: PathSource<'ast>,
         finalize: Finalize,
         record_partial_res: RecordPartialRes,
+        parent_qself: &Option<P<QSelf>>,
```

Comment:
Make this `Option<&QSelf>` and use `Some(&qself)` on the callsites, so then you don't need to do awkward things like unshadowing the variable with `qself_` and passing `&None` in places.